src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -432,4 +443,26 @@ private function loadFromExtensions($content)
             $this->container->loadFromExtension($namespace, $values);
         }
     }
+
+    /**
+     * Checks the keywords used to define a service.
+     *
+     * @param string $id         The service name
+     * @param array  $definition The service definition to check
+     *
+     * @throws InvalidArgumentException When an invalid keyword is used
+     */
+    private static function checkDefinition($id, array $definition)
+    {
+        foreach ($definition as $key => $value) {
+            if (!in_array($key, static::$keywords)) {
+                throw new InvalidArgumentException(sprintf(
```

Comment:
I've submitted the PR against `master` branch, which means the upcoming `3.1` version. So it doesn't affect previous versions like `2.8` where you can still use `factory_*` keywords.